sphinx_immaterial/inline_icons.py

```diff
@@ -9,22 +9,27 @@
 from sphinx.writers.html5 import HTML5Translator
 
 
-def read_svg_into_span(builder: Builder, icon_name: str, classes: List[str]) -> str:
-    """Reads the SVG data into a HTML span element."""
-    svg = Path(__file__).parent / ".icons" / icon_name
-    if not svg.exists():
-        static_paths: List[str] = getattr(builder.config, "html_static_path")
-        for path in static_paths:
-            svg = Path(builder.srcdir) / path / icon_name
-            if svg.exists():
-                break
-        else:
-            raise FileNotFoundError(
-                f"{icon_name} not found in html_static_path and not bundled with theme"
-            )
-    svg_data = svg.read_text(encoding="utf-8")
-    css_classes = " ".join(classes)
-    return f'<span class="{css_classes}">{svg_data}</span>'
+def load_svg_into_builder_env(builder: Builder, icon_name: str) -> str:
+    """Reads the SVG data into a builder env variable for CSS generation."""
+    css_icon_name = icon_name.replace("/", "_").replace("\\", "_")
+    custom_icons = getattr(builder.env, "sphinx_immaterial_custom_icons", {})
+    icon_name += ".svg"
+    if css_icon_name not in custom_icons:
+        svg = Path(__file__).parent / ".icons" / icon_name
+        if not svg.exists():
+            static_paths: List[str] = getattr(builder.config, "html_static_path")
```

Comment:
Since these icons don't actually need to be copied to the output directory, I think it would be better to create a new config option for specifying icon paths, rather than re-using `html_static_path`.  Additionally, It might be better to search these user-specified icon paths first, to allow the user to override builtin icons.

Reply:
I have this implemented, but our docs are using
```py
sphinx_immaterial_icon_path = html_static_path
```
because the custom icon (sphinx_logo.svg) is displayed using both `si-icon` role and the `image` directive for demonstration purposes.